app/src/main/java/com/mathewsachin/fategrandautomata/accessibility/AccessibilityGestures.kt

```diff
@@ -43,9 +56,83 @@ class AccessibilityGestures @Inject constructor(
         gesturePrefs.swipeWaitTime.wait()
     }
 
+    /**
+     * Android 8+ swipe is precise due to use of continued gestures.
+     *
+     * Instead of swiping the whole distance as a single gesture,
+     * it is split into multiple small swipes, which is similar to how events are sent if a real human is doing it.
+     * There is a finger down delay, followed by multiple small swipe events, followed by a finger lift delay.
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    suspend fun swipe8(start: Location, end: Location) {
+        val xDiff = (end.X - start.X).toFloat()
+        val yDiff = (end.Y - start.Y).toFloat()
+        val direction = atan2(xDiff, yDiff)
+        var distanceLeft = sqrt(xDiff.pow(2) + yDiff.pow(2))
+
+        val thresholdDistance = 5f
+        val swipeDuration = 1L
+
+        var from = start
+
+        val mouseDownPath = Path().moveTo(start)
+
+        var lastStroke = GestureDescription.StrokeDescription(
+            mouseDownPath,
+            0,
+            200L,
+            true
+        )
+
+        performGesture(lastStroke)
+
+        while (distanceLeft > 0) {
+            val distanceToScroll = minOf(thresholdDistance, distanceLeft)
```

Comment:
yup, like FGO lua does with `manualTouch`.

Reply:
That together with the comment about scrolling on different devices reminded me of the fact that the gift box script in fgolua scrolled differently on each device. But with the swipe multiplier, that should be a non-issue.

Reply:
I was hoping it would scroll around the same on Android 8+ devices though.
This is just like what a real user would do. A finger down, followed by multiple small swipe events, followed by a pause at the destination before we lift the finger.
I was thinking that the pause at the swipe end location would make it consistent across devices.

Reply:
I checked and my mid-range phone can handle 10pixels/1ms at a time pretty well and that is 2x the speed of this as expected.
But, I think extremely low-end phones might have a tough time with that.
Let me check around 7-8 pixels